frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -379,11 +383,24 @@ function computeComparisonPeriodsAgo({
     dateUnitSettings.dateUnit,
   ).toLowerCase();
 
-  const prevDate = moment
+  const computedPrevDate = moment
     .parseZone(nextDate)
     .subtract(dateUnitsAgo, dateUnitSettings.dateUnit)
     .format();
 
+  const rowPeriodsAgo = getRowOfPeriodsAgo({
+    prevDate: computedPrevDate,
+    dateUnit: dateUnitSettings.dateUnit,
+    dateUnitsAgo,
+    dimensionColIndex,
+    metricColIndex,
+    nextValueRowIndex,
+    rows,
+  });
+
+  const prevDate = !isEmpty(rowPeriodsAgo)
+    ? rowPeriodsAgo[dimensionColIndex]
+    : computedPrevDate;
```

Comment:
Since we no longer want the condensed versions of the dates and the dates for native queries may not align perfectly with the `dateUnit` interval, we should get the actual date and display that (so we are not displaying an incorrect date as in the image below).

**Prior to Change**
![Pasted image 20240131183512](https://github.com/metabase/metabase/assets/22608765/5d88d343-604f-46b1-bd19-51109ec6cbea)
- See that the display date is `December 31, 2021`, but should be `October 11, 2021`


**After Change**
![image](https://github.com/metabase/metabase/assets/22608765/b9c6aa8f-a398-4c19-a547-d1446399b7c8)